test/integration/route_middleware_test.rb

```diff
@@ -43,6 +43,24 @@ class RouteMiddlewareTest < ActionDispatch::IntegrationTest
     assert_equal('anybody out there?', @response.body)
   end
 
+  test 'Can downcase cyrillic paths correctly' do
+    RouteDowncaser.configuration do |config|
+      config.redirect = true
+    end
+    get "/#{URI.encode_www_form_component('ПИВО')}"
```

Comment:
Interesting! So this is really how those characters are encoded? In that case this looks solid, good work.

Reply:
Here are some refs:

* https://tools.ietf.org/html/rfc1738 URIs only include ascii
* https://www.w3.org/International/articles/idn-and-iri IRIs and internationalization in urls in general
* https://stackoverflow.com/questions/2742852/unicode-characters-in-urls
* https://stackoverflow.com/questions/1916684/cant-open-unicode-url-with-python